---
title: "Mutation_library_analysis"
author: "Mu-En Chung"
date: "07/07/2022"
output: md_document
# output:
#   word_document: default
#   html_document: default
#   pdf_document: default
---

[Introduction]                                                               
This tool aligns a set of sample DNA sequences to a reference sequence,      
finds the mutations in each sample sequences, and then export a csv file with
each row being each sample sequence and their amino acid substitutions.      
Author: Dr. Mu-En Chung 
Email: chungmuen@gmail.com

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

enter the path to the folder containing the sequencing results and the name of your WT sequence below:
```{r user_input}
seq_file_path = "./"                                                            
ref_seq_name = "ClyF"
```

install_and_import_packages:
```{r install_and_import_packages}

if (!requireNamespace("BiocManager", quietly = TRUE)){
   install.packages("BiocManager")
}
if (!requireNamespace("msa", quietly = TRUE)){
   BiocManager::install("msa")
}

##import package msa
library(msa)
```

```{r examples}
###run example provided by msa package
mySequenceFile <- system.file("examples", "exampleAA.fasta", package="msa")
mySequences <- readAAStringSet(mySequenceFile)
print("example AA seq")
mySequences
myFirstAlignment <- msa(mySequences)
print("example AA alignment")
myFirstAlignment

##run example DNA alignment
myDNAFile <- system.file("examples", "exampleDNA.fasta", package="msa")
mySequencesDNA <- readDNAStringSet(myDNAFile)
print("example DNA seq")
mySequencesDNA
exDNA_Alignment <- msa(mySequencesDNA)
print("example DNA alignment")
exDNA_Alignment
```

run your alignment
```{r run your alignment}
##run my alignment
file_path <- list.files(path = seq_file_path,pattern = 'fasta',full.names = T)
seq_result <- readDNAStringSet(file_path)
myDNA_Alignment <- msa(seq_result)

#The print() function provided by the msa package provides some ways for customizing the output, such as, showing the entire alignment split over multiple blocks of sub-sequences
print("my alignment")
print(myDNA_Alignment, show="complete")
```

from alignment, calculate the mutation profile in DNA level and in Amino acid level
```{r analysis}
#convert alignment to a list of vectors and matrix 
convert <- msaConvert(myDNA_Alignment,type= "bio3d::fasta")

alignment_mat <- convert$ali
alignment_df <- as.data.frame(alignment_mat)

alignment_df_dREF <- alignment_df[-grep(ref_seq_name,row.names(alignment_df)),]
alignment_df_dREF <- alignment_df_dREF[order(row.names(alignment_df_dREF)),]

alignment_df_reorder <- as.data.frame(rbind(alignment_df[grep(ref_seq_name,row.names(alignment_df)),],alignment_df_dREF))

#remove columns extruding the reference sequence
keep_columns <- as.data.frame(which(alignment_df_reorder[1,]!= "-"))
trim_before <- keep_columns[1,1]-1
trim_after <- keep_columns[length(keep_columns[,1]),1]+1
trim_alignment <- alignment_df_reorder[,-c(1:trim_before,trim_after:ncol(alignment_df_reorder))]
# trim_alignment[1,]

#identify the position of "-" in the reference 
reference_row <- trim_alignment[1,]
print("afer alignment, the positions of - in the reference sequence are:")
which(reference_row == "-")

#identify the mutations and create a matrix of mutations (0=mutation, 1=no mutation)
mutation_count <- trim_alignment

for (i in 1:nrow(trim_alignment)) 
{
  for (j in 1:ncol(trim_alignment)) 
  {
    if (trim_alignment[i, j] == trim_alignment[1, j]) 
    {
      mutation_count[i, j] <- 1
    } 
    else 
    {
      mutation_count[i, j] <- 0
    }
  }
}

################################################

#classfy the mutations and create a matrix of mutations
mutation_count_TvTs <- trim_alignment

for (i in 1:nrow(trim_alignment)) 
{
  for (j in 1:ncol(trim_alignment)) 
  {
    if (trim_alignment[i,j] == trim_alignment[1,j]) 
    {
      mutation_count_TvTs[i,j] <- "-"
    } 
    else if (trim_alignment[i,j] == "A")
    {
      if (trim_alignment[1,j] == "T")
      {
        mutation_count_TvTs[i, j] <- "TA"
      }
      else if (trim_alignment[1,j] == "C")
      {
        mutation_count_TvTs[i, j] <- "CA"
      }
      else if (trim_alignment[1,j] == "G")
      {
        mutation_count_TvTs[i, j] <- "GA"
      }
      else if (trim_alignment[1,j] == "-")
      {
        mutation_count_TvTs[i, j] <- "Ins"
      }
    }
    else if (trim_alignment[i,j] == "T")
    {
      if (trim_alignment[1,j] == "A")
      {
        mutation_count_TvTs[i, j] <- "AT"
      }
      else if (trim_alignment[1,j] == "C")
      {
        mutation_count_TvTs[i, j] <- "CT"
      }
      else if (trim_alignment[1,j] == "G")
      {
        mutation_count_TvTs[i, j] <- "GT"
      }
      else if (trim_alignment[1,j] == "-")
      {
        mutation_count_TvTs[i, j] <- "Ins"
      }
    }
    else if (trim_alignment[i,j] == "C")
    {
      if (trim_alignment[1,j] == "A")
      {
        mutation_count_TvTs[i, j] <- "AC"
      }
      else if (trim_alignment[1,j] == "T")
      {
        mutation_count_TvTs[i, j] <- "TC"
      }
      else if (trim_alignment[1,j] == "G")
      {
        mutation_count_TvTs[i, j] <- "GC"
      }
      else if (trim_alignment[1,j] == "-")
      {
        mutation_count_TvTs[i, j] <- "Ins"
      }
    }
    else if (trim_alignment[i,j] == "G")
    {
      if (trim_alignment[1,j] == "A")
      {
        mutation_count_TvTs[i, j] <- "AG"
      }
      else if (trim_alignment[1,j] == "T")
      {
        mutation_count_TvTs[i, j] <- "TG"
      }
      else if (trim_alignment[1,j] == "C")
      {
        mutation_count_TvTs[i, j] <- "CG"
      }
      else if (trim_alignment[1,j] == "-")
      {
        mutation_count_TvTs[i, j] <- "Ins"
      }
    }
    else if (trim_alignment[i,j] == "-")
    {
      mutation_count_TvTs[i, j] <- "Del"
    }  
    else
    {
      mutation_count_TvTs[i, j] <- "?"
    }  
  }
}

#check if all the elements are assigned as "-", "S", "V", "Del", "Ins", "?"
print('the number of  "-", "S", "V", "Del", "Ins", "?":')
for(i in 1:nrow(mutation_count_TvTs))
{
  a <- table(as.character(mutation_count_TvTs[i,]))
  print(row.names(mutation_count_TvTs)[i])
  print(a)
}

#show row names
rownames <- as.data.frame(row.names(mutation_count))

#count the mutations in each sequence
#output: 
#id - S V Ins Del ?
mutation_count_row <- t(as.matrix(table(as.character(mutation_count_TvTs[1,]))))
addition <- matrix(c(0,0,0,0,0,0,0,0,0,0,0,0,0,0,0), nrow = 1, ncol = 15)
mutation_count_row <- as.data.frame(cbind(mutation_count_row,addition))
colnames(mutation_count_row) <- c("-","AG", "GA", "AT", "TA", "AC", "CA", "GC", "CG", "GT", "TG", "CT", "TC", "Del", "Ins", "?")

for(i in 2:nrow(mutation_count_TvTs))
{
  mutation_count_row2 <- as.data.frame(t(as.matrix(table(as.character(mutation_count_TvTs[i,])))))
  if(!('AG' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'AG'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('GA' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'GA'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('AT' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'AT'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('TA' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'TA'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('AC' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'AC'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('CA' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'CA'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('GC' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'GC'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('CG' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'CG'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('GT' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'GT'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('TG' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'TG'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('CT' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'CT'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('TC' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'TC'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('Del' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'Del'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('Ins' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- 'Ins'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  if(!('?' %in% names(mutation_count_row2)))
  {
    addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
    colnames(addition) <- '?'
    mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
  }
  mutation_count_row2
  
  mutation_count_row <- rbind(mutation_count_row,mutation_count_row2)
}
row.names(mutation_count_row) <- row.names(mutation_count_TvTs)

################################################
#identify the row with the most mutations
n_mut <- sum(mutation_count[1,]==0)
for(i in 2:nrow(mutation_count))
{
  n_mut2 <- sum(mutation_count[i,]==0)
  if(n_mut2 >= n_mut)
  {
    n_mut <- n_mut2
  }
}
###############################################
#identify mutation locations
mutation_locations <- rep(NA,n_mut)

for(i in 2:nrow(mutation_count))
{
  mutation_locations2 <- rep(NA,n_mut)
  mutations <- t(as.data.frame(which(mutation_count[i,]==0)))
  for (j in 1:n_mut)
  {
    if (j <= ncol(mutations))
    {
      mutation_locations2 [j] <- mutations[1,j]
    }
  }
  mutation_locations <-  rbind(mutation_locations, mutation_locations2)
}
row.names(mutation_locations) <- row.names(mutation_count)

###########################################################################
#identify the amino acid mutations
if(sum(trim_alignment[1,]=="-")!= 0)
{
  trim_dash <- trim_alignment[,-c(which(trim_alignment[1,]=="-"))] 
}else
{
  trim_dash <- trim_alignment
}

modulo_mut_loc_df <- mutation_locations 
AAmutations_df <- mutation_locations
i <- 8
for(i in 1:nrow(mutation_locations))
{
  mutation_location_location <- which(mutation_locations[i,] != "NA")
  if(length(mutation_location_location) > 0)
  {
    for(j in 1:length(mutation_location_location))
    {
      print("-----start------")
      print("i=")
      print(i)
      print("j=")
      print(j)
      print("mutation_locations[i,j]=")
      print(mutation_locations[i,j])
      mutation_location_x <- as.numeric(mutation_locations[i,j]) 
      print("mutation_location_x=")
      print(mutation_location_x)
      mutation_location_remove_dash <- mutation_location_x - 
        sum(trim_alignment[1,1:(mutation_location_x-1)] == "-")
      print("mutation_location_remove_dash=")
      print(mutation_location_remove_dash)
      
      if(trim_alignment[1,mutation_location_x] == "-")
      {
        # decide the insertion location in AA seq
        modulo_mut_loc <- (mutation_location_remove_dash-1) %% 3
        modulo_mut_loc_df[i,j] <- modulo_mut_loc
        # if the insertion happen in between two codons, the mutation is counted to sit in the later codon.  
        if (modulo_mut_loc == 0){
          codon_start <- mutation_location_remove_dash
          codon_end <- mutation_location_remove_dash+2
          MUTcodon <- trim_alignment[i,mutation_location_x:mutation_location_x+3]
          aa_position <- (mutation_location_remove_dash-1)/3+1
        }
        if (modulo_mut_loc == 1){
          codon_start <- mutation_location_remove_dash-1
          codon_end <- mutation_location_remove_dash+1
          MUTcodon <- trim_alignment[i,(mutation_location_x-1):(mutation_location_x+2)]
          aa_position <- ceiling((mutation_location_remove_dash-1)/3)
        }
        if (modulo_mut_loc == 2){
          codon_start <- mutation_location_remove_dash-2
          codon_end <- mutation_location_remove_dash
          MUTcodon <- trim_alignment[i,(mutation_location_x-2):(mutation_location_x+1)]
          aa_position <- ceiling((mutation_location_remove_dash-1)/3)
        }
        
        print("frame_shift_ins")
        MUTaa <- "frame_shift_ins"

      }else{
        modulo_mut_loc <- mutation_location_remove_dash %% 3
        modulo_mut_loc_df[i,j] <- modulo_mut_loc
        
        if (modulo_mut_loc == 0){
          codon_start <- mutation_location_remove_dash-2
          codon_end <- mutation_location_remove_dash
        }
        if (modulo_mut_loc == 1){
          codon_start <- mutation_location_remove_dash
          codon_end <- mutation_location_remove_dash+2
        }
        if (modulo_mut_loc == 2){
          codon_start <- mutation_location_remove_dash-1
          codon_end <- mutation_location_remove_dash+1
        }
        aa_position <- ceiling(mutation_location_remove_dash/3)
        MUTcodon <- c(trim_dash[i,codon_start:codon_end])
        
        if (sum(MUTcodon[]== "-") != 0)
        {
          MUTaa <- "frame_shift_del"
          print("frame_shift_del")
        }
        
        else if (sum(MUTcodon[] %in% c("A","T","C","G")) == 3)
        {
          MUTcodon_DNAstr <- DNAString(paste(MUTcodon, collapse = ""))
          print("MUTcodon_DNAstr")
          print(MUTcodon_DNAstr)
          MUTaa <- translate(MUTcodon_DNAstr)
          print("translation!MUTaa")
          
        }
        
        else if (sum(MUTcodon[] %in% c("A","T","C","G","-")) != 3)
        {
          MUTaa <- "unidentifiable base"
          print("unidentifiable base :(")
        }
      }
      print("aaPosition")
      print(aa_position)
      
      WTcodon <- paste(c(trim_dash[1,codon_start:codon_end]), collapse="")
      WTcodon_DNAstr <- DNAString(WTcodon)
      WTaa <- translate(WTcodon_DNAstr)
      print("WTcodon and aa")
      print(WTcodon_DNAstr)
      print(WTaa)
      print("MUTcodon and aa")
      print(MUTcodon)
      print(MUTaa)
      
      WTaa_cha <- as.character(WTaa)
      MUTaa_cha <- as.character(MUTaa)
      aa_position_cha <- as.character(aa_position)
      AAmutations <- c(WTaa_cha,aa_position_cha,MUTaa_cha)
      AAmutations_df[i,j] <- paste(AAmutations, collapse="")
      print("-----END-----")
    }
  }
}

```

show results: AAmutations_df
```{r display_AAmutations_df}
AAmutations_df
```

Show results: DNA mutation location (alignment location)
```{r display_mutation_locations}
mutation_locations
```

Show results: DNA mutation counts.
-: no mutation
AG: A to G
GA: G to A
Del: Deletion
Ins: Insertion
?: Not recognized nucleotide
```{r display_mutation_count_row}
mutation_count_row
```


export the result tables as csv
```{r export_results}
##turn "NA" to nothing for output csv
AAmutations_removeNA <- AAmutations_df
AAmutations_removeNA[is.na(AAmutations_removeNA)] <- ""
mutation_locations_removeNA <- mutation_locations
mutation_locations_removeNA[is.na(mutation_locations_removeNA)] <- ""
##create a new folder in the current working directory 
dir.create("mutation_analysis_results", recursive = T)

##Save the dataframes 
write.csv(mutation_count_row,"mutation_analysis_results/DNAmutation_counts.csv", row.names = TRUE)
write.csv(mutation_locations_removeNA,"mutation_analysis_results/DNAmutation_locations.csv", row.names = TRUE)
write.csv(AAmutations_removeNA,"mutation_analysis_results/AAmutations.csv", row.names = TRUE)

list.files('mutation_analysis_results',full.names = T)
```
