\[Introduction\]  
This tool aligns a set of sample DNA sequences to a reference
sequence,  
finds the mutations in each sample sequences, and then export a csv file
with each row being each sample sequence and their amino acid
substitutions.  
Author: Dr.Â Mu-En Chung Email: <chungmuen@gmail.com>

enter the path to the folder containing the sequencing results and the
name of your WT sequence below:

    seq_file_path = "./"                                                            
    ref_seq_name = "ClyF"

install\_and\_import\_packages:

    if (!requireNamespace("BiocManager", quietly = TRUE)){
       install.packages("BiocManager")
    }
    if (!requireNamespace("msa", quietly = TRUE)){
       BiocManager::install("msa")
    }

    ##import package msa
    library(msa)

    ## Loading required package: Biostrings

    ## Loading required package: BiocGenerics

    ## 
    ## Attaching package: 'BiocGenerics'

    ## The following objects are masked from 'package:stats':
    ## 
    ##     IQR, mad, sd, var, xtabs

    ## The following objects are masked from 'package:base':
    ## 
    ##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    ##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    ##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    ##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    ##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    ##     union, unique, unsplit, which.max, which.min

    ## Loading required package: S4Vectors

    ## Loading required package: stats4

    ## 
    ## Attaching package: 'S4Vectors'

    ## The following objects are masked from 'package:base':
    ## 
    ##     expand.grid, I, unname

    ## Loading required package: IRanges

    ## Loading required package: XVector

    ## Loading required package: GenomeInfoDb

    ## 
    ## Attaching package: 'Biostrings'

    ## The following object is masked from 'package:base':
    ## 
    ##     strsplit

    ###run example provided by msa package
    mySequenceFile <- system.file("examples", "exampleAA.fasta", package="msa")
    mySequences <- readAAStringSet(mySequenceFile)
    print("example AA seq")

    ## [1] "example AA seq"

    mySequences

    ## AAStringSet object of length 9:
    ##     width seq                                               names               
    ## [1]   452 MSTAVLENPGLGRKLSDFGQETS...LKILADSINSEIGILCSALQKIK PH4H_Homo_sapiens
    ## [2]   453 MAAVVLENGVLSRKLSDFGQETS...KILADSINSEVGILCNALQKIKS PH4H_Rattus_norve...
    ## [3]   453 MAAVVLENGVLSRKLSDFGQETS...KILADSINSEVGILCHALQKIKS PH4H_Mus_musculus
    ## [4]   297 MNDRADFVVPDITTRKNVGLSHD...DVAPDDLVLNAGDRQGWADTEDV PH4H_Chromobacter...
    ## [5]   262 MKTTQYVARQPDDNGFIHYPETE...MALVHEAMRLGLHAPLFPPKQAA PH4H_Pseudomonas_...
    ## [6]   451 MSALVLESRALGRKLSDFGQETS...LKILADSISSEVEILCSALQKLK PH4H_Bos_taurus
    ## [7]   313 MAIATPTSAAPTPAPAGFTGTLT...DVVDGDAVLNAGTREGWADTADI PH4H_Ralstonia_so...
    ## [8]   294 MSGDGLSNGPPPGARPDWTIDQG...AVLTRGTQAYATAGGRLAGAAAG PH4H_Caulobacter_...
    ## [9]   275 MSVAEYARDCAAQGLRGDYSVCR...QTADFEAIVARRKDQKALDPATV PH4H_Rhizobium_loti

    myFirstAlignment <- msa(mySequences)

    ## use default substitution matrix

    print("example AA alignment")

    ## [1] "example AA alignment"

    myFirstAlignment

    ## CLUSTAL 2.1  
    ## 
    ## Call:
    ##    msa(mySequences)
    ## 
    ## MsaAAMultipleAlignment with 9 rows and 456 columns
    ##     aln                                                    names
    ## [1] MAAVVLENGVLSRKLSDFGQETSYIE...QLKILADSINSEVGILCNALQKIKS PH4H_Rattus_norve...
    ## [2] MAAVVLENGVLSRKLSDFGQETSYIE...QLKILADSINSEVGILCHALQKIKS PH4H_Mus_musculus
    ## [3] MSTAVLENPGLGRKLSDFGQETSYIE...QLKILADSINSEIGILCSALQKIK- PH4H_Homo_sapiens
    ## [4] MSALVLESRALGRKLSDFGQETSYIE...QLKILADSISSEVEILCSALQKLK- PH4H_Bos_taurus
    ## [5] --------------------------...LNAGDRQGWADTEDV---------- PH4H_Chromobacter...
    ## [6] --------------------------...LNAGTREGWADTADI---------- PH4H_Ralstonia_so...
    ## [7] --------------------------...LTRGT-QAYATAGGRLAGAAAG--- PH4H_Caulobacter_...
    ## [8] --------------------------...------------------------- PH4H_Pseudomonas_...
    ## [9] --------------------------...------------------------- PH4H_Rhizobium_loti
    ## Con --------------------------...??????????????IL??A???--- Consensus

    ##run example DNA alignment
    myDNAFile <- system.file("examples", "exampleDNA.fasta", package="msa")
    mySequencesDNA <- readDNAStringSet(myDNAFile)
    print("example DNA seq")

    ## [1] "example DNA seq"

    mySequencesDNA

    ## DNAStringSet object of length 3:
    ##     width seq                                               names               
    ## [1]  2680 CAGCTGGGGGTAAGGGGGGCGGA...AGTAATAAAACATTAGTAGTTTA PAH_Homo_sapiens
    ## [2]  2152 AGTCCAGGACTGGAGTTGGGTAT...CATTATATAAAACAAAAAAAAAA PAH_Mus_musculus
    ## [3]  2040 CAAGTTAAAACCATCAGCCCTCC...AAAAAAAAAAAAAAAAAAAAAAA PAH_Rattus_norveg...

    exDNA_Alignment <- msa(mySequencesDNA)

    ## use default substitution matrix

    print("example DNA alignment")

    ## [1] "example DNA alignment"

    exDNA_Alignment

    ## CLUSTAL 2.1  
    ## 
    ## Call:
    ##    msa(mySequencesDNA)
    ## 
    ## MsaDNAMultipleAlignment with 3 rows and 2716 columns
    ##     aln                                                    names
    ## [1] --------------------------...------------------------- PAH_Mus_musculus
    ## [2] --------------------------...------------------------- PAH_Rattus_norveg...
    ## [3] CAGCTGGGGGTAAGGGGGGCGGATTA...TTAGTAATAAAACATTAGTAGTTTA PAH_Homo_sapiens
    ## Con --------------------------...------------------------- Consensus

run your alignment

    ##run my alignment
    file_path <- list.files(path = seq_file_path,pattern = 'fasta',full.names = T)
    seq_result <- readDNAStringSet(file_path)
    myDNA_Alignment <- msa(seq_result)

    ## use default substitution matrix

    #The print() function provided by the msa package provides some ways for customizing the output, such as, showing the entire alignment split over multiple blocks of sub-sequences
    print("my alignment")

    ## [1] "my alignment"

    print(myDNA_Alignment, show="complete")

    ## 
    ## MsaDNAMultipleAlignment with 28 rows and 1296 columns
    ##      aln (1..53)                                           names
    ##  [1] ------------------------------------------------AAAAT 31_T7 Biotec_CGN
    ##  [2] -------------------------------------------------GGAT 33_T7 Biotec_CGN
    ##  [3] ------------------------------------------------GAAGT 43_T7 Biotec_CGN
    ##  [4] -----------------------------------------------GGAAAA 21_T7 Biotec_CGN
    ##  [5] ------------------------------------------------GAAAT 51_T7 Biotec_CGN
    ##  [6] ------------------------------------------------AGGTT 14_T7 Biotec_CGN
    ##  [7] -AAGGGGAAATTCCTCTAGAAATAAATTTTTGTTAACTTTAAGAAGGAGATAT 55_T7 Biotec_CGN
    ##  [8] --GGGGGAAATTCCTCTAGAAATAAATTTTTGTTA-CTTTAAGAAGGAGATAT 56_T7 Biotec_CGN
    ##  [9] --GGGGGAAATTCCTCTAGAAATAA-TTTTTGTTA-CTTTAAGAAGGAGATAT 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] -CGGGGGAAATTTCTCTAGAAATAA-TTTTTGTTA-CTTTAAGAAGGAGATAT 23_T7 Biotec_CGN
    ## [22] ---GGGGAAATTTCTCTAGAAATAA-TTTTTGTTA-CTTTAAGAAGGAGATAT 24_T7 Biotec_CGN
    ## [23] --AGGGGAAATTCCTCTAGAAATAA-TTTTTGTTA-CTTTAAGAAGGAGATAT 25_T7 Biotec_CGN
    ## [24] CAAGGGGAAATTCCTCTAGAAATAATTTTT-GTTTACTTTAAGAAGGAGATAT 32_T7 Biotec_CGN
    ## [25] -------------------AAATAATTTTTTGTTTACTTTAAGAAGGAGATAT 34_T7 Biotec_CGN
    ## [26] --AGGGGAAATTCCTCTAGAAATAA-TTTTTGTTA-CTTTAAGAAGGAGATAT 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] ----------------------------------------------GAAGT-T 45_T7 Biotec_CGN
    ##  Con --?GGGGAAATTCCTCTAGAAATAA-TTTTTGTTA-CTTTAAGAAGGAGATAT Consensus 
    ## 
    ##      aln (54..106)                                         names
    ##  [1] TCCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 31_T7 Biotec_CGN
    ##  [2] TACATGCATCACCATCACCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 33_T7 Biotec_CGN
    ##  [3] TACATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 43_T7 Biotec_CGN
    ##  [4] TACATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 21_T7 Biotec_CGN
    ##  [5] TCCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 51_T7 Biotec_CGN
    ##  [6] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 14_T7 Biotec_CGN
    ##  [7] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 55_T7 Biotec_CGN
    ##  [8] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 56_T7 Biotec_CGN
    ##  [9] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 23_T7 Biotec_CGN
    ## [22] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 24_T7 Biotec_CGN
    ## [23] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 25_T7 Biotec_CGN
    ## [24] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 32_T7 Biotec_CGN
    ## [25] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 34_T7 Biotec_CGN
    ## [26] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 35_T7 Biotec_CGN
    ## [27] ---ATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC ClyF
    ## [28] ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC 45_T7 Biotec_CGN
    ##  Con ACCATGCATCACCATCATCACCACGCCCTTCCGAAAACGGGGAAACCTACAGC Consensus 
    ## 
    ##      aln (107..159)                                        names
    ##  [1] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 31_T7 Biotec_CGN
    ##  [2] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 33_T7 Biotec_CGN
    ##  [3] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 43_T7 Biotec_CGN
    ##  [4] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 21_T7 Biotec_CGN
    ##  [5] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 51_T7 Biotec_CGN
    ##  [6] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 14_T7 Biotec_CGN
    ##  [7] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 55_T7 Biotec_CGN
    ##  [8] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 56_T7 Biotec_CGN
    ##  [9] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 23_T7 Biotec_CGN
    ## [22] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 24_T7 Biotec_CGN
    ## [23] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 25_T7 Biotec_CGN
    ## [24] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 32_T7 Biotec_CGN
    ## [25] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 34_T7 Biotec_CGN
    ## [26] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 35_T7 Biotec_CGN
    ## [27] AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG ClyF
    ## [28] AAGGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG 45_T7 Biotec_CGN
    ##  Con AAAGCAAGTGGTGGACTGGGCAATCAACTTGATCGGCTCCGGTGTTGACGTTG Consensus 
    ## 
    ##      aln (160..212)                                        names
    ##  [1] GCGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 31_T7 Biotec_CGN
    ##  [2] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 33_T7 Biotec_CGN
    ##  [3] ACGGGTATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 43_T7 Biotec_CGN
    ##  [4] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 21_T7 Biotec_CGN
    ##  [5] ACGGATATTACGGTCGTCAAAGTTGGGACTTGCCAAACTATATCTTTAACCGT 51_T7 Biotec_CGN
    ##  [6] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 14_T7 Biotec_CGN
    ##  [7] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 55_T7 Biotec_CGN
    ##  [8] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 56_T7 Biotec_CGN
    ##  [9] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 23_T7 Biotec_CGN
    ## [22] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 24_T7 Biotec_CGN
    ## [23] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 25_T7 Biotec_CGN
    ## [24] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 32_T7 Biotec_CGN
    ## [25] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 34_T7 Biotec_CGN
    ## [26] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTCTAACCGT 35_T7 Biotec_CGN
    ## [27] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT ClyF
    ## [28] ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT 45_T7 Biotec_CGN
    ##  Con ACGGATATTACGGTCGTCAATGTTGGGACTTGCCAAACTATATCTTTAACCGT Consensus 
    ## 
    ##      aln (213..265)                                        names
    ##  [1] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 31_T7 Biotec_CGN
    ##  [2] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 33_T7 Biotec_CGN
    ##  [3] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 43_T7 Biotec_CGN
    ##  [4] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 21_T7 Biotec_CGN
    ##  [5] TACTGGAATTTCAAGACCCCAGGAAATACTCGCGACATGGCTTGGTATCGTTA 51_T7 Biotec_CGN
    ##  [6] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 14_T7 Biotec_CGN
    ##  [7] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 55_T7 Biotec_CGN
    ##  [8] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 56_T7 Biotec_CGN
    ##  [9] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 23_T7 Biotec_CGN
    ## [22] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 24_T7 Biotec_CGN
    ## [23] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 25_T7 Biotec_CGN
    ## [24] TACTGGAATTTCGAGACCCCAGGAAATGCCCGCGACATGGCTTGGTATCGTTA 32_T7 Biotec_CGN
    ## [25] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 34_T7 Biotec_CGN
    ## [26] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 35_T7 Biotec_CGN
    ## [27] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA ClyF
    ## [28] TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA 45_T7 Biotec_CGN
    ##  Con TACTGGAATTTCAAGACCCCAGGAAATGCTCGCGACATGGCTTGGTATCGTTA Consensus 
    ## 
    ##      aln (266..318)                                        names
    ##  [1] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 31_T7 Biotec_CGN
    ##  [2] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 33_T7 Biotec_CGN
    ##  [3] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 43_T7 Biotec_CGN
    ##  [4] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 21_T7 Biotec_CGN
    ##  [5] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCGGCGATTTCGTACCAAAACCAG 51_T7 Biotec_CGN
    ##  [6] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 14_T7 Biotec_CGN
    ##  [7] CCCAGAGGGTTTCAAAGTTCTCCGCAACACCAGCGATTTCGTACCAAAACCAG 55_T7 Biotec_CGN
    ##  [8] CCCAGAGGGTTTCAAAGTTCTCCGCAACACCAGCGATTTCGTACCAAAACCAG 56_T7 Biotec_CGN
    ##  [9] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 23_T7 Biotec_CGN
    ## [22] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 24_T7 Biotec_CGN
    ## [23] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 25_T7 Biotec_CGN
    ## [24] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTATCAAAACCAG 32_T7 Biotec_CGN
    ## [25] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 34_T7 Biotec_CGN
    ## [26] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 35_T7 Biotec_CGN
    ## [27] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG ClyF
    ## [28] CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG 45_T7 Biotec_CGN
    ##  Con CCCAGAGGGTTTCAAAGTTTTCCGCAACACCAGCGATTTCGTACCAAAACCAG Consensus 
    ## 
    ##      aln (319..371)                                        names
    ##  [1] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 31_T7 Biotec_CGN
    ##  [2] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 33_T7 Biotec_CGN
    ##  [3] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 43_T7 Biotec_CGN
    ##  [4] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 21_T7 Biotec_CGN
    ##  [5] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 51_T7 Biotec_CGN
    ##  [6] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 14_T7 Biotec_CGN
    ##  [7] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 55_T7 Biotec_CGN
    ##  [8] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 56_T7 Biotec_CGN
    ##  [9] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] GCGACATCGCGGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 23_T7 Biotec_CGN
    ## [22] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 24_T7 Biotec_CGN
    ## [23] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 25_T7 Biotec_CGN
    ## [24] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 32_T7 Biotec_CGN
    ## [25] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 34_T7 Biotec_CGN
    ## [26] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC 35_T7 Biotec_CGN
    ## [27] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC ClyF
    ## [28] GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGAGGCAC 45_T7 Biotec_CGN
    ##  Con GCGACATCGCAGTGTGGACTGGGGGGAACTATAACTGGAATACATGGGGGCAC Consensus 
    ## 
    ##      aln (372..424)                                        names
    ##  [1] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 31_T7 Biotec_CGN
    ##  [2] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 33_T7 Biotec_CGN
    ##  [3] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 43_T7 Biotec_CGN
    ##  [4] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 21_T7 Biotec_CGN
    ##  [5] ACAGGA-TCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 51_T7 Biotec_CGN
    ##  [6] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 14_T7 Biotec_CGN
    ##  [7] ACAGGAATCGTCGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 55_T7 Biotec_CGN
    ##  [8] ACAGGAATCGTCGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 56_T7 Biotec_CGN
    ##  [9] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 23_T7 Biotec_CGN
    ## [22] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 24_T7 Biotec_CGN
    ## [23] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 25_T7 Biotec_CGN
    ## [24] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGGCCA 32_T7 Biotec_CGN
    ## [25] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 34_T7 Biotec_CGN
    ## [26] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 35_T7 Biotec_CGN
    ## [27] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA ClyF
    ## [28] ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA 45_T7 Biotec_CGN
    ##  Con ACAGGAATCGTTGTTGGCCCAAGCACGAAGTCTTACTTCTATTCGGTTGACCA Consensus 
    ## 
    ##      aln (425..477)                                        names
    ##  [1] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 31_T7 Biotec_CGN
    ##  [2] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 33_T7 Biotec_CGN
    ##  [3] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 43_T7 Biotec_CGN
    ##  [4] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 21_T7 Biotec_CGN
    ##  [5] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 51_T7 Biotec_CGN
    ##  [6] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 14_T7 Biotec_CGN
    ##  [7] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 55_T7 Biotec_CGN
    ##  [8] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 56_T7 Biotec_CGN
    ##  [9] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 23_T7 Biotec_CGN
    ## [22] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 24_T7 Biotec_CGN
    ## [23] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 25_T7 Biotec_CGN
    ## [24] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 32_T7 Biotec_CGN
    ## [25] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 34_T7 Biotec_CGN
    ## [26] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 35_T7 Biotec_CGN
    ## [27] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC ClyF
    ## [28] AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC 45_T7 Biotec_CGN
    ##  Con AAATTGGAACAACAGTAACAGTTATGTCGGCAGTCCTGCCGCCAAAATCAAGC Consensus 
    ## 
    ##      aln (478..530)                                        names
    ##  [1] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 31_T7 Biotec_CGN
    ##  [2] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 33_T7 Biotec_CGN
    ##  [3] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 43_T7 Biotec_CGN
    ##  [4] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 21_T7 Biotec_CGN
    ##  [5] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 51_T7 Biotec_CGN
    ##  [6] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 14_T7 Biotec_CGN
    ##  [7] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 55_T7 Biotec_CGN
    ##  [8] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 56_T7 Biotec_CGN
    ##  [9] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 23_T7 Biotec_CGN
    ## [22] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 24_T7 Biotec_CGN
    ## [23] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 25_T7 Biotec_CGN
    ## [24] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 32_T7 Biotec_CGN
    ## [25] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 34_T7 Biotec_CGN
    ## [26] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 35_T7 Biotec_CGN
    ## [27] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG ClyF
    ## [28] ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG 45_T7 Biotec_CGN
    ##  Con ACTCCTACTTTGGAGTGACCCATTTTGTGCGTCCCGCTTACAAAGCCGAACCG Consensus 
    ## 
    ##      aln (531..583)                                        names
    ##  [1] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 31_T7 Biotec_CGN
    ##  [2] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 33_T7 Biotec_CGN
    ##  [3] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 43_T7 Biotec_CGN
    ##  [4] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 21_T7 Biotec_CGN
    ##  [5] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 51_T7 Biotec_CGN
    ##  [6] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 14_T7 Biotec_CGN
    ##  [7] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 55_T7 Biotec_CGN
    ##  [8] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 56_T7 Biotec_CGN
    ##  [9] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 23_T7 Biotec_CGN
    ## [22] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 24_T7 Biotec_CGN
    ## [23] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 25_T7 Biotec_CGN
    ## [24] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 32_T7 Biotec_CGN
    ## [25] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 34_T7 Biotec_CGN
    ## [26] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 35_T7 Biotec_CGN
    ## [27] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC ClyF
    ## [28] AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC 45_T7 Biotec_CGN
    ##  Con AAGCCTACTCCGCCATCTCGCAGTCGTTCATACCGCGAAACAGGAACGATGAC Consensus 
    ## 
    ##      aln (584..636)                                        names
    ##  [1] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 31_T7 Biotec_CGN
    ##  [2] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 33_T7 Biotec_CGN
    ##  [3] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 43_T7 Biotec_CGN
    ##  [4] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 21_T7 Biotec_CGN
    ##  [5] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 51_T7 Biotec_CGN
    ##  [6] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 14_T7 Biotec_CGN
    ##  [7] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 55_T7 Biotec_CGN
    ##  [8] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 56_T7 Biotec_CGN
    ##  [9] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 23_T7 Biotec_CGN
    ## [22] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 24_T7 Biotec_CGN
    ## [23] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 25_T7 Biotec_CGN
    ## [24] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 32_T7 Biotec_CGN
    ## [25] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 34_T7 Biotec_CGN
    ## [26] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 35_T7 Biotec_CGN
    ## [27] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA ClyF
    ## [28] GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA 45_T7 Biotec_CGN
    ##  Con GGTCACGGTGGATGCGCTGAACGTACGTCGCGCTCCGAACACCTCGGGTGAAA Consensus 
    ## 
    ##      aln (637..689)                                        names
    ##  [1] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCGTCATC 31_T7 Biotec_CGN
    ##  [2] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 33_T7 Biotec_CGN
    ##  [3] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 43_T7 Biotec_CGN
    ##  [4] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 21_T7 Biotec_CGN
    ##  [5] TTGTGGCTGTGTACAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 51_T7 Biotec_CGN
    ##  [6] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 14_T7 Biotec_CGN
    ##  [7] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 55_T7 Biotec_CGN
    ##  [8] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 56_T7 Biotec_CGN
    ##  [9] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 23_T7 Biotec_CGN
    ## [22] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 24_T7 Biotec_CGN
    ## [23] TTGTGGCTGTGTTTAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 25_T7 Biotec_CGN
    ## [24] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 32_T7 Biotec_CGN
    ## [25] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 34_T7 Biotec_CGN
    ## [26] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 35_T7 Biotec_CGN
    ## [27] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC ClyF
    ## [28] TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC 45_T7 Biotec_CGN
    ##  Con TTGTGGCTGTGTATAAACGCGGTGAGAGCTTCGACTATGATACGGTCATCATC Consensus 
    ## 
    ##      aln (690..742)                                        names
    ##  [1] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 31_T7 Biotec_CGN
    ##  [2] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 33_T7 Biotec_CGN
    ##  [3] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 43_T7 Biotec_CGN
    ##  [4] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 21_T7 Biotec_CGN
    ##  [5] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGGAGCGCAA 51_T7 Biotec_CGN
    ##  [6] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 14_T7 Biotec_CGN
    ##  [7] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 55_T7 Biotec_CGN
    ##  [8] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 56_T7 Biotec_CGN
    ##  [9] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 23_T7 Biotec_CGN
    ## [22] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 24_T7 Biotec_CGN
    ## [23] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 25_T7 Biotec_CGN
    ## [24] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 32_T7 Biotec_CGN
    ## [25] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 34_T7 Biotec_CGN
    ## [26] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 35_T7 Biotec_CGN
    ## [27] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA ClyF
    ## [28] GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA 45_T7 Biotec_CGN
    ##  Con GACGTGAATGGTTACGTCTGGGTGTCTTACATTGGCGGGTCCGGGAAGCGCAA Consensus 
    ## 
    ##      aln (743..795)                                        names
    ##  [1] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 31_T7 Biotec_CGN
    ##  [2] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 33_T7 Biotec_CGN
    ##  [3] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 43_T7 Biotec_CGN
    ##  [4] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 21_T7 Biotec_CGN
    ##  [5] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 51_T7 Biotec_CGN
    ##  [6] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 14_T7 Biotec_CGN
    ##  [7] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 55_T7 Biotec_CGN
    ##  [8] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 56_T7 Biotec_CGN
    ##  [9] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 23_T7 Biotec_CGN
    ## [22] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 24_T7 Biotec_CGN
    ## [23] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 25_T7 Biotec_CGN
    ## [24] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 32_T7 Biotec_CGN
    ## [25] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 34_T7 Biotec_CGN
    ## [26] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 35_T7 Biotec_CGN
    ## [27] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG ClyF
    ## [28] CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG 45_T7 Biotec_CGN
    ##  Con CTACGTTGCAACAGGGGCAACGAAGGACGGCAAGCGTTTTGGAAATGCATGGG Consensus 
    ## 
    ##      aln (796..848)                                        names
    ##  [1] G-TACTTTTAAGTAATAAGAA-------------------------------- 31_T7 Biotec_CGN
    ##  [2] G-TACTTTTAAGTAATAAGAA-------------------------------- 33_T7 Biotec_CGN
    ##  [3] G-TACTTTTAAGTAATAAGAA-------------------------------- 43_T7 Biotec_CGN
    ##  [4] G-TACTTTTAAGTAATAAGAA-------------------------------- 21_T7 Biotec_CGN
    ##  [5] G-TACTTTTAAGTAATAAGAA-------------------------------- 51_T7 Biotec_CGN
    ##  [6] G-TACTTTTAAGTAATAAGAA-------------------------------- 14_T7 Biotec_CGN
    ##  [7] G-TACTTTTAAGTAATAAGAA-------------------------------- 55_T7 Biotec_CGN
    ##  [8] G-TACTTTTAAGTAATAAGAA-------------------------------- 56_T7 Biotec_CGN
    ##  [9] G-TACTTTTAAGTAATAAGAA-------------------------------- 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] G-TACTTTTAAGTAATAAGAA-------------------------------- 23_T7 Biotec_CGN
    ## [22] G-TACTTTTAAGTAATAAGAA-------------------------------- 24_T7 Biotec_CGN
    ## [23] G-TACTTTTAAGTAATAAGAA-------------------------------- 25_T7 Biotec_CGN
    ## [24] G-TACTTTTAAGTAATAAGAA-------------------------------- 32_T7 Biotec_CGN
    ## [25] G-TACTTTTAAGTAATAAGAA-------------------------------- 34_T7 Biotec_CGN
    ## [26] G-TACTTTTAAGTAATAAGAA-------------------------------- 35_T7 Biotec_CGN
    ## [27] G-TACTTTTAAGTAA-------------------------------------- ClyF
    ## [28] G-TACTTTTAAGTAATAAGAATTCGAGCTCCGTCGACAAGCTTGCGGGCCGCA 45_T7 Biotec_CGN
    ##  Con G-TACTTTTAAGTAATAAGAA-------------------------------- Consensus 
    ## 
    ##      aln (849..901)                                        names
    ##  [1] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 31_T7 Biotec_CGN
    ##  [2] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 33_T7 Biotec_CGN
    ##  [3] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 43_T7 Biotec_CGN
    ##  [4] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 21_T7 Biotec_CGN
    ##  [5] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 51_T7 Biotec_CGN
    ##  [6] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 14_T7 Biotec_CGN
    ##  [7] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 55_T7 Biotec_CGN
    ##  [8] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 56_T7 Biotec_CGN
    ##  [9] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 23_T7 Biotec_CGN
    ## [22] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 24_T7 Biotec_CGN
    ## [23] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 25_T7 Biotec_CGN
    ## [24] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 32_T7 Biotec_CGN
    ## [25] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 34_T7 Biotec_CGN
    ## [26] TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] CTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC 45_T7 Biotec_CGN
    ##  Con TTCGAGCTCCGTCGACAAGCTTGCGGCCGCACTCGAGCACCACCACC-ACCAC Consensus 
    ## 
    ##      aln (902..954)                                        names
    ##  [1] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 31_T7 Biotec_CGN
    ##  [2] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 33_T7 Biotec_CGN
    ##  [3] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 43_T7 Biotec_CGN
    ##  [4] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 21_T7 Biotec_CGN
    ##  [5] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 51_T7 Biotec_CGN
    ##  [6] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 14_T7 Biotec_CGN
    ##  [7] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 55_T7 Biotec_CGN
    ##  [8] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 56_T7 Biotec_CGN
    ##  [9] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 23_T7 Biotec_CGN
    ## [22] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 24_T7 Biotec_CGN
    ## [23] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 25_T7 Biotec_CGN
    ## [24] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 32_T7 Biotec_CGN
    ## [25] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 34_T7 Biotec_CGN
    ## [26] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT 45_T7 Biotec_CGN
    ##  Con CACTGA-GATCCGG-CTGCTAACAAAGCCC---GAAAGGAAGCTGAGTTGGCT Consensus 
    ## 
    ##      aln (955..1007)                                       names
    ##  [1] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 31_T7 Biotec_CGN
    ##  [2] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 33_T7 Biotec_CGN
    ##  [3] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 43_T7 Biotec_CGN
    ##  [4] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 21_T7 Biotec_CGN
    ##  [5] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 51_T7 Biotec_CGN
    ##  [6] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 14_T7 Biotec_CGN
    ##  [7] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 55_T7 Biotec_CGN
    ##  [8] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 56_T7 Biotec_CGN
    ##  [9] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 23_T7 Biotec_CGN
    ## [22] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 24_T7 Biotec_CGN
    ## [23] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 25_T7 Biotec_CGN
    ## [24] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 32_T7 Biotec_CGN
    ## [25] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 34_T7 Biotec_CGN
    ## [26] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA 45_T7 Biotec_CGN
    ##  Con GCTGCCACC-GCTGAGCAATAACT--AGCATAACCCCTT--GGGGCCTCTAAA Consensus 
    ## 
    ##      aln (1008..1060)                                      names
    ##  [1] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 31_T7 Biotec_CGN
    ##  [2] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 33_T7 Biotec_CGN
    ##  [3] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 43_T7 Biotec_CGN
    ##  [4] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 21_T7 Biotec_CGN
    ##  [5] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 51_T7 Biotec_CGN
    ##  [6] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 14_T7 Biotec_CGN
    ##  [7] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 55_T7 Biotec_CGN
    ##  [8] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 56_T7 Biotec_CGN
    ##  [9] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 23_T7 Biotec_CGN
    ## [22] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 24_T7 Biotec_CGN
    ## [23] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 25_T7 Biotec_CGN
    ## [24] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 32_T7 Biotec_CGN
    ## [25] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 34_T7 Biotec_CGN
    ## [26] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT 45_T7 Biotec_CGN
    ##  Con C--GGGTCTTGAGGGG---TTTTTTGCTGAAA--GGAGGAACTATATCCGGAT Consensus 
    ## 
    ##      aln (1061..1113)                                      names
    ##  [1] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 31_T7 Biotec_CGN
    ##  [2] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 33_T7 Biotec_CGN
    ##  [3] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 43_T7 Biotec_CGN
    ##  [4] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 21_T7 Biotec_CGN
    ##  [5] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 51_T7 Biotec_CGN
    ##  [6] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 14_T7 Biotec_CGN
    ##  [7] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 55_T7 Biotec_CGN
    ##  [8] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 56_T7 Biotec_CGN
    ##  [9] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 23_T7 Biotec_CGN
    ## [22] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 24_T7 Biotec_CGN
    ## [23] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 25_T7 Biotec_CGN
    ## [24] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 32_T7 Biotec_CGN
    ## [25] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 34_T7 Biotec_CGN
    ## [26] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG 45_T7 Biotec_CGN
    ##  Con TGGCGAATGGGACGCGCCCTGTAGCGGCGCATTAAGCGCGGCGGGTGTGGTGG Consensus 
    ## 
    ##      aln (1114..1166)                                      names
    ##  [1] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 31_T7 Biotec_CGN
    ##  [2] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 33_T7 Biotec_CGN
    ##  [3] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 43_T7 Biotec_CGN
    ##  [4] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 21_T7 Biotec_CGN
    ##  [5] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 51_T7 Biotec_CGN
    ##  [6] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 14_T7 Biotec_CGN
    ##  [7] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 55_T7 Biotec_CGN
    ##  [8] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 56_T7 Biotec_CGN
    ##  [9] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 23_T7 Biotec_CGN
    ## [22] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 24_T7 Biotec_CGN
    ## [23] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 25_T7 Biotec_CGN
    ## [24] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 32_T7 Biotec_CGN
    ## [25] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 34_T7 Biotec_CGN
    ## [26] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC 45_T7 Biotec_CGN
    ##  Con TTACGCGCAGCGTGACCGCTACACTTGCCAGCGCCCTAGCGCCCGCTCCTTTC Consensus 
    ## 
    ##      aln (1167..1219)                                      names
    ##  [1] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 31_T7 Biotec_CGN
    ##  [2] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 33_T7 Biotec_CGN
    ##  [3] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 43_T7 Biotec_CGN
    ##  [4] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 21_T7 Biotec_CGN
    ##  [5] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 51_T7 Biotec_CGN
    ##  [6] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 14_T7 Biotec_CGN
    ##  [7] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGG-CTTTCCCCGTCAAGCTC 55_T7 Biotec_CGN
    ##  [8] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGGTTTTCCCCGTCAAGCTC 56_T7 Biotec_CGN
    ##  [9] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCT---------------- 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGGTTTTCCCCGTCAAGCTC 23_T7 Biotec_CGN
    ## [22] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGGTTTTCCCCGTCAAGCTC 24_T7 Biotec_CGN
    ## [23] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGGTTTTCCCCGTCAAGCTC 25_T7 Biotec_CGN
    ## [24] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGGTTTTCCCCGTCAAGCTC 32_T7 Biotec_CGN
    ## [25] GCTTTCTTCCCTTCCTTTCTCGC------------------------------ 34_T7 Biotec_CGN
    ## [26] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTCT 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGGCTTTCCCCGTCAAGCTC- 45_T7 Biotec_CGN
    ##  Con GCTTTCTTCCCTTCCTTTCTCGCCACGTTCGCCGG?TTT?CCC????A???C? Consensus 
    ## 
    ##      aln (1220..1272)                                      names
    ##  [1] AAATCGGGGGCTCCCTTTAGGGTTCCGATTTAGTGCTTTACGGCACCTCGACC 31_T7 Biotec_CGN
    ##  [2] AAATCGGGGGCTCCCTTTAGGGTTCCAATTTAGTGCTTTACGGCACCTCGACC 33_T7 Biotec_CGN
    ##  [3] AAATCGGGGGCTCCCTTTAGGGTTCCAATTTATTGCTTTACGGCACCTC---- 43_T7 Biotec_CGN
    ##  [4] AAATCGGGGGCTCCCTTTAGGGTTCCAATT-AGTGCTTTACGGCCC------- 21_T7 Biotec_CGN
    ##  [5] AAATCGGGGGCTCCCTTTAGGGTTCCAATTTAATGCTTTACGGCACCTC---- 51_T7 Biotec_CGN
    ##  [6] AAATCGGGGGCTCCCTTTAGGGTTCCAATTTA--------------------- 14_T7 Biotec_CGN
    ##  [7] TAAATCGGGGGCTCC-------------------------------------- 55_T7 Biotec_CGN
    ##  [8] TAAATCGGGGGCTCCCTTTAGGGTTCCCAATTTA------------------- 56_T7 Biotec_CGN
    ##  [9] ----------------------------------------------------- 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] TAAATCGGGGGCTCCCTTTAGGGTTCCA------------------------- 23_T7 Biotec_CGN
    ## [22] TAAATCGGGGGCTCCCTTTAGGGTTCC-------------------------- 24_T7 Biotec_CGN
    ## [23] TAAATCGGGGGCTCCCTTTAGGGTTCCAATTTA-------------------- 25_T7 Biotec_CGN
    ## [24] TAAATCGGGGGCTCCCTTTAGGGTTCCGAATTTAGG----------------- 32_T7 Biotec_CGN
    ## [25] ----------------------------------------------------- 34_T7 Biotec_CGN
    ## [26] AAATCGGGGGCTCCCTTTAGGGTTCCAA------------------------- 35_T7 Biotec_CGN
    ## [27] ----------------------------------------------------- ClyF
    ## [28] ----------------------------------------------------- 45_T7 Biotec_CGN
    ##  Con ?AA???GGGG??-??-------------------------------------- Consensus 
    ## 
    ##      aln (1273..1296)         names
    ##  [1] CCAAAAACTTGATTAGGGTGAGGG 31_T7 Biotec_CGN
    ##  [2] CCAAAAACTTGATTAGGGT----- 33_T7 Biotec_CGN
    ##  [3] ------------------------ 43_T7 Biotec_CGN
    ##  [4] ------------------------ 21_T7 Biotec_CGN
    ##  [5] ------------------------ 51_T7 Biotec_CGN
    ##  [6] ------------------------ 14_T7 Biotec_CGN
    ##  [7] ------------------------ 55_T7 Biotec_CGN
    ##  [8] ------------------------ 56_T7 Biotec_CGN
    ##  [9] ------------------------ 13_T7 Biotec_CGN 
    ##  ... ...
    ## [21] ------------------------ 23_T7 Biotec_CGN
    ## [22] ------------------------ 24_T7 Biotec_CGN
    ## [23] ------------------------ 25_T7 Biotec_CGN
    ## [24] ------------------------ 32_T7 Biotec_CGN
    ## [25] ------------------------ 34_T7 Biotec_CGN
    ## [26] ------------------------ 35_T7 Biotec_CGN
    ## [27] ------------------------ ClyF
    ## [28] ------------------------ 45_T7 Biotec_CGN
    ##  Con ------------------------ Consensus

from alignment, calculate the mutation profile in DNA level and in Amino
acid level

    #convert alignment to a list of vectors and matrix 
    convert <- msaConvert(myDNA_Alignment,type= "bio3d::fasta")

    alignment_mat <- convert$ali
    alignment_df <- as.data.frame(alignment_mat)

    alignment_df_dREF <- alignment_df[-grep(ref_seq_name,row.names(alignment_df)),]
    alignment_df_dREF <- alignment_df_dREF[order(row.names(alignment_df_dREF)),]

    alignment_df_reorder <- as.data.frame(rbind(alignment_df[grep(ref_seq_name,row.names(alignment_df)),],alignment_df_dREF))

    #remove columns extruding the reference sequence
    keep_columns <- as.data.frame(which(alignment_df_reorder[1,]!= "-"))
    trim_before <- keep_columns[1,1]-1
    trim_after <- keep_columns[length(keep_columns[,1]),1]+1
    trim_alignment <- alignment_df_reorder[,-c(1:trim_before,trim_after:ncol(alignment_df_reorder))]
    # trim_alignment[1,]

    #identify the position of "-" in the reference 
    reference_row <- trim_alignment[1,]
    print("afer alignment, the positions of - in the reference sequence are:")

    ## [1] "afer alignment, the positions of - in the reference sequence are:"

    which(reference_row == "-")

    ## [1] 741

    #identify the mutations and create a matrix of mutations (0=mutation, 1=no mutation)
    mutation_count <- trim_alignment

    for (i in 1:nrow(trim_alignment)) 
    {
      for (j in 1:ncol(trim_alignment)) 
      {
        if (trim_alignment[i, j] == trim_alignment[1, j]) 
        {
          mutation_count[i, j] <- 1
        } 
        else 
        {
          mutation_count[i, j] <- 0
        }
      }
    }

    ################################################

    #classfy the mutations and create a matrix of mutations
    mutation_count_TvTs <- trim_alignment

    for (i in 1:nrow(trim_alignment)) 
    {
      for (j in 1:ncol(trim_alignment)) 
      {
        if (trim_alignment[i,j] == trim_alignment[1,j]) 
        {
          mutation_count_TvTs[i,j] <- "-"
        } 
        else if (trim_alignment[i,j] == "A")
        {
          if (trim_alignment[1,j] == "T")
          {
            mutation_count_TvTs[i, j] <- "TA"
          }
          else if (trim_alignment[1,j] == "C")
          {
            mutation_count_TvTs[i, j] <- "CA"
          }
          else if (trim_alignment[1,j] == "G")
          {
            mutation_count_TvTs[i, j] <- "GA"
          }
          else if (trim_alignment[1,j] == "-")
          {
            mutation_count_TvTs[i, j] <- "Ins"
          }
        }
        else if (trim_alignment[i,j] == "T")
        {
          if (trim_alignment[1,j] == "A")
          {
            mutation_count_TvTs[i, j] <- "AT"
          }
          else if (trim_alignment[1,j] == "C")
          {
            mutation_count_TvTs[i, j] <- "CT"
          }
          else if (trim_alignment[1,j] == "G")
          {
            mutation_count_TvTs[i, j] <- "GT"
          }
          else if (trim_alignment[1,j] == "-")
          {
            mutation_count_TvTs[i, j] <- "Ins"
          }
        }
        else if (trim_alignment[i,j] == "C")
        {
          if (trim_alignment[1,j] == "A")
          {
            mutation_count_TvTs[i, j] <- "AC"
          }
          else if (trim_alignment[1,j] == "T")
          {
            mutation_count_TvTs[i, j] <- "TC"
          }
          else if (trim_alignment[1,j] == "G")
          {
            mutation_count_TvTs[i, j] <- "GC"
          }
          else if (trim_alignment[1,j] == "-")
          {
            mutation_count_TvTs[i, j] <- "Ins"
          }
        }
        else if (trim_alignment[i,j] == "G")
        {
          if (trim_alignment[1,j] == "A")
          {
            mutation_count_TvTs[i, j] <- "AG"
          }
          else if (trim_alignment[1,j] == "T")
          {
            mutation_count_TvTs[i, j] <- "TG"
          }
          else if (trim_alignment[1,j] == "C")
          {
            mutation_count_TvTs[i, j] <- "CG"
          }
          else if (trim_alignment[1,j] == "-")
          {
            mutation_count_TvTs[i, j] <- "Ins"
          }
        }
        else if (trim_alignment[i,j] == "-")
        {
          mutation_count_TvTs[i, j] <- "Del"
        }  
        else
        {
          mutation_count_TvTs[i, j] <- "?"
        }  
      }
    }

    #check if all the elements are assigned as "-", "S", "V", "Del", "Ins", "?"
    print('the number of  "-", "S", "V", "Del", "Ins", "?":')

    ## [1] "the number of  \"-\", \"S\", \"V\", \"Del\", \"Ins\", \"?\":"

    for(i in 1:nrow(mutation_count_TvTs))
    {
      a <- table(as.character(mutation_count_TvTs[i,]))
      print(row.names(mutation_count_TvTs)[i])
      print(a)
    }

    ## [1] "ClyF"
    ## 
    ##   - 
    ## 754 
    ## [1] "11_T7 Biotec_CGN"
    ## 
    ##   -  AG 
    ## 753   1 
    ## [1] "12_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "13_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "14_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "15_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "21_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "22_T7 Biotec_CGN"
    ## 
    ##   - Ins 
    ## 753   1 
    ## [1] "23_T7 Biotec_CGN"
    ## 
    ##   -  AG 
    ## 753   1 
    ## [1] "24_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "25_T7 Biotec_CGN"
    ## 
    ##   -  AT 
    ## 753   1 
    ## [1] "31_T7 Biotec_CGN"
    ## 
    ##   -  AG 
    ## 752   2 
    ## [1] "32_T7 Biotec_CGN"
    ## 
    ##   -  AG  CT  TC 
    ## 750   2   1   1 
    ## [1] "33_T7 Biotec_CGN"
    ## 
    ##   -  TC 
    ## 753   1 
    ## [1] "34_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "35_T7 Biotec_CGN"
    ## 
    ##   -  TC 
    ## 753   1 
    ## [1] "41_T7 Biotec_CGN"
    ## 
    ##   -  CT 
    ## 753   1 
    ## [1] "42_T7 Biotec_CGN"
    ## 
    ##   -  CT  GA  TC 
    ## 751   1   1   1 
    ## [1] "43_T7 Biotec_CGN"
    ## 
    ##   -  AG 
    ## 753   1 
    ## [1] "44_T7 Biotec_CGN"
    ## 
    ##   -  TA  TC 
    ## 751   1   2 
    ## [1] "45_T7 Biotec_CGN"
    ## 
    ##   -  AG  GA 
    ## 752   1   1 
    ## [1] "46_T7 Biotec_CGN"
    ## 
    ##   -  AG  GA  TC 
    ## 751   1   1   1 
    ## [1] "51_T7 Biotec_CGN"
    ## 
    ##   -  AG Del  GA  TA  TC 
    ## 748   2   1   1   1   1 
    ## [1] "52_T7 Biotec_CGN"
    ## 
    ##   - 
    ## 754 
    ## [1] "53_T7 Biotec_CGN"
    ## 
    ##   -  AG  CT  TC 
    ## 750   1   2   1 
    ## [1] "54_T7 Biotec_CGN"
    ## 
    ##   -  AG  TC 
    ## 751   1   2 
    ## [1] "55_T7 Biotec_CGN"
    ## 
    ##   -  TC 
    ## 752   2 
    ## [1] "56_T7 Biotec_CGN"
    ## 
    ##   -  TC 
    ## 752   2

    #show row names
    rownames <- as.data.frame(row.names(mutation_count))

    #count the mutations in each sequence
    #output: 
    #id - S V Ins Del ?
    mutation_count_row <- t(as.matrix(table(as.character(mutation_count_TvTs[1,]))))
    addition <- matrix(c(0,0,0,0,0,0,0,0,0,0,0,0,0,0,0), nrow = 1, ncol = 15)
    mutation_count_row <- as.data.frame(cbind(mutation_count_row,addition))
    colnames(mutation_count_row) <- c("-","AG", "GA", "AT", "TA", "AC", "CA", "GC", "CG", "GT", "TG", "CT", "TC", "Del", "Ins", "?")

    for(i in 2:nrow(mutation_count_TvTs))
    {
      mutation_count_row2 <- as.data.frame(t(as.matrix(table(as.character(mutation_count_TvTs[i,])))))
      if(!('AG' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'AG'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('GA' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'GA'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('AT' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'AT'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('TA' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'TA'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('AC' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'AC'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('CA' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'CA'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('GC' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'GC'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('CG' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'CG'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('GT' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'GT'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('TG' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'TG'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('CT' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'CT'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('TC' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'TC'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('Del' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'Del'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('Ins' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- 'Ins'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      if(!('?' %in% names(mutation_count_row2)))
      {
        addition <- as.data.frame(matrix(0, nrow = 1, ncol = 1))
        colnames(addition) <- '?'
        mutation_count_row2 <- as.data.frame(cbind(mutation_count_row2,addition))
      }
      mutation_count_row2
      
      mutation_count_row <- rbind(mutation_count_row,mutation_count_row2)
    }
    row.names(mutation_count_row) <- row.names(mutation_count_TvTs)

    ################################################
    #identify the row with the most mutations
    n_mut <- sum(mutation_count[1,]==0)
    for(i in 2:nrow(mutation_count))
    {
      n_mut2 <- sum(mutation_count[i,]==0)
      if(n_mut2 >= n_mut)
      {
        n_mut <- n_mut2
      }
    }
    ###############################################
    #identify mutation locations
    mutation_locations <- rep(NA,n_mut)

    for(i in 2:nrow(mutation_count))
    {
      mutation_locations2 <- rep(NA,n_mut)
      mutations <- t(as.data.frame(which(mutation_count[i,]==0)))
      for (j in 1:n_mut)
      {
        if (j <= ncol(mutations))
        {
          mutation_locations2 [j] <- mutations[1,j]
        }
      }
      mutation_locations <-  rbind(mutation_locations, mutation_locations2)
    }
    row.names(mutation_locations) <- row.names(mutation_count)

    ###########################################################################
    #identify the amino acid mutations
    if(sum(trim_alignment[1,]=="-")!= 0)
    {
      trim_dash <- trim_alignment[,-c(which(trim_alignment[1,]=="-"))] 
    }else
    {
      trim_dash <- trim_alignment
    }

    modulo_mut_loc_df <- mutation_locations 
    AAmutations_df <- mutation_locations
    i <- 8
    for(i in 1:nrow(mutation_locations))
    {
      mutation_location_location <- which(mutation_locations[i,] != "NA")
      if(length(mutation_location_location) > 0)
      {
        for(j in 1:length(mutation_location_location))
        {
          print("-----start------")
          print("i=")
          print(i)
          print("j=")
          print(j)
          print("mutation_locations[i,j]=")
          print(mutation_locations[i,j])
          mutation_location_x <- as.numeric(mutation_locations[i,j]) 
          print("mutation_location_x=")
          print(mutation_location_x)
          mutation_location_remove_dash <- mutation_location_x - 
            sum(trim_alignment[1,1:(mutation_location_x-1)] == "-")
          print("mutation_location_remove_dash=")
          print(mutation_location_remove_dash)
          
          if(trim_alignment[1,mutation_location_x] == "-")
          {
            # decide the insertion location in AA seq
            modulo_mut_loc <- (mutation_location_remove_dash-1) %% 3
            modulo_mut_loc_df[i,j] <- modulo_mut_loc
            # if the insertion happen in between two codons, the mutation is counted to sit in the later codon.  
            if (modulo_mut_loc == 0){
              codon_start <- mutation_location_remove_dash
              codon_end <- mutation_location_remove_dash+2
              MUTcodon <- trim_alignment[i,mutation_location_x:mutation_location_x+3]
              aa_position <- (mutation_location_remove_dash-1)/3+1
            }
            if (modulo_mut_loc == 1){
              codon_start <- mutation_location_remove_dash-1
              codon_end <- mutation_location_remove_dash+1
              MUTcodon <- trim_alignment[i,(mutation_location_x-1):(mutation_location_x+2)]
              aa_position <- ceiling((mutation_location_remove_dash-1)/3)
            }
            if (modulo_mut_loc == 2){
              codon_start <- mutation_location_remove_dash-2
              codon_end <- mutation_location_remove_dash
              MUTcodon <- trim_alignment[i,(mutation_location_x-2):(mutation_location_x+1)]
              aa_position <- ceiling((mutation_location_remove_dash-1)/3)
            }
            
            print("frame_shift_ins")
            MUTaa <- "frame_shift_ins"

          }else{
            modulo_mut_loc <- mutation_location_remove_dash %% 3
            modulo_mut_loc_df[i,j] <- modulo_mut_loc
            
            if (modulo_mut_loc == 0){
              codon_start <- mutation_location_remove_dash-2
              codon_end <- mutation_location_remove_dash
            }
            if (modulo_mut_loc == 1){
              codon_start <- mutation_location_remove_dash
              codon_end <- mutation_location_remove_dash+2
            }
            if (modulo_mut_loc == 2){
              codon_start <- mutation_location_remove_dash-1
              codon_end <- mutation_location_remove_dash+1
            }
            aa_position <- ceiling(mutation_location_remove_dash/3)
            MUTcodon <- c(trim_dash[i,codon_start:codon_end])
            
            if (sum(MUTcodon[]== "-") != 0)
            {
              MUTaa <- "frame_shift_del"
              print("frame_shift_del")
            }
            
            else if (sum(MUTcodon[] %in% c("A","T","C","G")) == 3)
            {
              MUTcodon_DNAstr <- DNAString(paste(MUTcodon, collapse = ""))
              print("MUTcodon_DNAstr")
              print(MUTcodon_DNAstr)
              MUTaa <- translate(MUTcodon_DNAstr)
              print("translation!MUTaa")
              
            }
            
            else if (sum(MUTcodon[] %in% c("A","T","C","G","-")) != 3)
            {
              MUTaa <- "unidentifiable base"
              print("unidentifiable base :(")
            }
          }
          print("aaPosition")
          print(aa_position)
          
          WTcodon <- paste(c(trim_dash[1,codon_start:codon_end]), collapse="")
          WTcodon_DNAstr <- DNAString(WTcodon)
          WTaa <- translate(WTcodon_DNAstr)
          print("WTcodon and aa")
          print(WTcodon_DNAstr)
          print(WTaa)
          print("MUTcodon and aa")
          print(MUTcodon)
          print(MUTaa)
          
          WTaa_cha <- as.character(WTaa)
          MUTaa_cha <- as.character(MUTaa)
          aa_position_cha <- as.character(aa_position)
          AAmutations <- c(WTaa_cha,aa_position_cha,MUTaa_cha)
          AAmutations_df[i,j] <- paste(AAmutations, collapse="")
          print("-----END-----")
        }
      }
    }

    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 2
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 11_T7 Biotec_CGN 
    ##              181 
    ## [1] "mutation_location_x="
    ## [1] 181
    ## [1] "mutation_location_remove_dash="
    ## [1] 181
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GAT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 61
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AAT
    ## 1-letter AAString object
    ## seq: N
    ## [1] "MUTcodon and aa"
    ## $V237
    ## [1] "G"
    ## 
    ## $V238
    ## [1] "A"
    ## 
    ## $V239
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: D
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 8
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 22_T7 Biotec_CGN 
    ##              741 
    ## [1] "mutation_location_x="
    ## [1] 741
    ## [1] "mutation_location_remove_dash="
    ## [1] 741
    ## [1] "frame_shift_ins"
    ## [1] "aaPosition"
    ## [1] 247
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GGT
    ## 1-letter AAString object
    ## seq: G
    ## [1] "MUTcodon and aa"
    ##                  V795 V796 V797 V798
    ## 22_T7 Biotec_CGN    G    G    G    T
    ## [1] "frame_shift_ins"
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 9
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 23_T7 Biotec_CGN 
    ##              273 
    ## [1] "mutation_location_x="
    ## [1] 273
    ## [1] "mutation_location_remove_dash="
    ## [1] 273
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GCG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 91
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GCA
    ## 1-letter AAString object
    ## seq: A
    ## [1] "MUTcodon and aa"
    ## $V327
    ## [1] "G"
    ## 
    ## $V328
    ## [1] "C"
    ## 
    ## $V329
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: A
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 11
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 25_T7 Biotec_CGN 
    ##              593 
    ## [1] "mutation_location_x="
    ## [1] 593
    ## [1] "mutation_location_remove_dash="
    ## [1] 593
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: TTT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 198
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TAT
    ## 1-letter AAString object
    ## seq: Y
    ## [1] "MUTcodon and aa"
    ## $V648
    ## [1] "T"
    ## 
    ## $V649
    ## [1] "T"
    ## 
    ## $V650
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: F
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 12
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 31_T7 Biotec_CGN 
    ##              104 
    ## [1] "mutation_location_x="
    ## [1] 104
    ## [1] "mutation_location_remove_dash="
    ## [1] 104
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 35
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GAC
    ## 1-letter AAString object
    ## seq: D
    ## [1] "MUTcodon and aa"
    ## $V159
    ## [1] "G"
    ## 
    ## $V160
    ## [1] "G"
    ## 
    ## $V161
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: G
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 12
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 31_T7 Biotec_CGN 
    ##              628 
    ## [1] "mutation_location_x="
    ## [1] 628
    ## [1] "mutation_location_remove_dash="
    ## [1] 628
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GTC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 210
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: ATC
    ## 1-letter AAString object
    ## seq: I
    ## [1] "MUTcodon and aa"
    ## $V684
    ## [1] "G"
    ## 
    ## $V685
    ## [1] "T"
    ## 
    ## $V686
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: V
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 13
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 32_T7 Biotec_CGN 
    ##              169 
    ## [1] "mutation_location_x="
    ## [1] 169
    ## [1] "mutation_location_remove_dash="
    ## [1] 169
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GAG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 57
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AAG
    ## 1-letter AAString object
    ## seq: K
    ## [1] "MUTcodon and aa"
    ## $V225
    ## [1] "G"
    ## 
    ## $V226
    ## [1] "A"
    ## 
    ## $V227
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: E
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 13
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 32_T7 Biotec_CGN 
    ##              186 
    ## [1] "mutation_location_x="
    ## [1] 186
    ## [1] "mutation_location_remove_dash="
    ## [1] 186
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GCC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 62
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GCT
    ## 1-letter AAString object
    ## seq: A
    ## [1] "MUTcodon and aa"
    ## $V240
    ## [1] "G"
    ## 
    ## $V241
    ## [1] "C"
    ## 
    ## $V242
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: A
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 13
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 32_T7 Biotec_CGN 
    ##              253 
    ## [1] "mutation_location_x="
    ## [1] 253
    ## [1] "mutation_location_remove_dash="
    ## [1] 253
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: TCA
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 85
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CCA
    ## 1-letter AAString object
    ## seq: P
    ## [1] "MUTcodon and aa"
    ## $V309
    ## [1] "T"
    ## 
    ## $V310
    ## [1] "C"
    ## 
    ## $V311
    ## [1] "A"
    ## 
    ## 1-letter AAString object
    ## seq: S
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 13
    ## [1] "j="
    ## [1] 4
    ## [1] "mutation_locations[i,j]="
    ## 32_T7 Biotec_CGN 
    ##              365 
    ## [1] "mutation_location_x="
    ## [1] 365
    ## [1] "mutation_location_remove_dash="
    ## [1] 365
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 122
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GAC
    ## 1-letter AAString object
    ## seq: D
    ## [1] "MUTcodon and aa"
    ## $V420
    ## [1] "G"
    ## 
    ## $V421
    ## [1] "G"
    ## 
    ## $V422
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: G
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 14
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 33_T7 Biotec_CGN 
    ##               15 
    ## [1] "mutation_location_x="
    ## [1] 15
    ## [1] "mutation_location_remove_dash="
    ## [1] 15
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CAC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 5
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CAT
    ## 1-letter AAString object
    ## seq: H
    ## [1] "MUTcodon and aa"
    ## $V69
    ## [1] "C"
    ## 
    ## $V70
    ## [1] "A"
    ## 
    ## $V71
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: H
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 16
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 35_T7 Biotec_CGN 
    ##              149 
    ## [1] "mutation_location_x="
    ## [1] 149
    ## [1] "mutation_location_remove_dash="
    ## [1] 149
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: TCT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 50
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TTT
    ## 1-letter AAString object
    ## seq: F
    ## [1] "MUTcodon and aa"
    ## $V204
    ## [1] "T"
    ## 
    ## $V205
    ## [1] "C"
    ## 
    ## $V206
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: S
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 17
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 41_T7 Biotec_CGN 
    ##               29 
    ## [1] "mutation_location_x="
    ## [1] 29
    ## [1] "mutation_location_remove_dash="
    ## [1] 29
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CTG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 10
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CCG
    ## 1-letter AAString object
    ## seq: P
    ## [1] "MUTcodon and aa"
    ## $V84
    ## [1] "C"
    ## 
    ## $V85
    ## [1] "T"
    ## 
    ## $V86
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: M
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 18
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 42_T7 Biotec_CGN 
    ##               90 
    ## [1] "mutation_location_x="
    ## [1] 90
    ## [1] "mutation_location_remove_dash="
    ## [1] 90
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: TCT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 30
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TCC
    ## 1-letter AAString object
    ## seq: S
    ## [1] "MUTcodon and aa"
    ## $V144
    ## [1] "T"
    ## 
    ## $V145
    ## [1] "C"
    ## 
    ## $V146
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: S
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 18
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 42_T7 Biotec_CGN 
    ##              501 
    ## [1] "mutation_location_x="
    ## [1] 501
    ## [1] "mutation_location_remove_dash="
    ## [1] 501
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 167
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CGT
    ## 1-letter AAString object
    ## seq: R
    ## [1] "MUTcodon and aa"
    ## $V555
    ## [1] "C"
    ## 
    ## $V556
    ## [1] "G"
    ## 
    ## $V557
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: R
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 18
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 42_T7 Biotec_CGN 
    ##              543 
    ## [1] "mutation_location_x="
    ## [1] 543
    ## [1] "mutation_location_remove_dash="
    ## [1] 543
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GCA
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 181
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GCG
    ## 1-letter AAString object
    ## seq: A
    ## [1] "MUTcodon and aa"
    ## $V597
    ## [1] "G"
    ## 
    ## $V598
    ## [1] "C"
    ## 
    ## $V599
    ## [1] "A"
    ## 
    ## 1-letter AAString object
    ## seq: A
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 19
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 43_T7 Biotec_CGN 
    ##              108 
    ## [1] "mutation_location_x="
    ## [1] 108
    ## [1] "mutation_location_remove_dash="
    ## [1] 108
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GGG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 36
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GGA
    ## 1-letter AAString object
    ## seq: G
    ## [1] "MUTcodon and aa"
    ## $V162
    ## [1] "G"
    ## 
    ## $V163
    ## [1] "G"
    ## 
    ## $V164
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: G
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 20
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 44_T7 Biotec_CGN 
    ##              117 
    ## [1] "mutation_location_x="
    ## [1] 117
    ## [1] "mutation_location_remove_dash="
    ## [1] 117
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 39
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GGT
    ## 1-letter AAString object
    ## seq: G
    ## [1] "MUTcodon and aa"
    ## $V171
    ## [1] "G"
    ## 
    ## $V172
    ## [1] "G"
    ## 
    ## $V173
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: G
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 20
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 44_T7 Biotec_CGN 
    ##              126 
    ## [1] "mutation_location_x="
    ## [1] 126
    ## [1] "mutation_location_remove_dash="
    ## [1] 126
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: TGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 42
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TGT
    ## 1-letter AAString object
    ## seq: C
    ## [1] "MUTcodon and aa"
    ## $V180
    ## [1] "T"
    ## 
    ## $V181
    ## [1] "G"
    ## 
    ## $V182
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: C
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 20
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 44_T7 Biotec_CGN 
    ##              251 
    ## [1] "mutation_location_x="
    ## [1] 251
    ## [1] "mutation_location_remove_dash="
    ## [1] 251
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GAA
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 84
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GTA
    ## 1-letter AAString object
    ## seq: V
    ## [1] "MUTcodon and aa"
    ## $V306
    ## [1] "G"
    ## 
    ## $V307
    ## [1] "A"
    ## 
    ## $V308
    ## [1] "A"
    ## 
    ## 1-letter AAString object
    ## seq: E
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 21
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 45_T7 Biotec_CGN 
    ##               53 
    ## [1] "mutation_location_x="
    ## [1] 53
    ## [1] "mutation_location_remove_dash="
    ## [1] 53
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: AGG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 18
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AAG
    ## 1-letter AAString object
    ## seq: K
    ## [1] "MUTcodon and aa"
    ## $V108
    ## [1] "A"
    ## 
    ## $V109
    ## [1] "G"
    ## 
    ## $V110
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: R
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 21
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 45_T7 Biotec_CGN 
    ##              310 
    ## [1] "mutation_location_x="
    ## [1] 310
    ## [1] "mutation_location_remove_dash="
    ## [1] 310
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: AGG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 104
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GGG
    ## 1-letter AAString object
    ## seq: G
    ## [1] "MUTcodon and aa"
    ## $V366
    ## [1] "A"
    ## 
    ## $V367
    ## [1] "G"
    ## 
    ## $V368
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: R
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 22
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 46_T7 Biotec_CGN 
    ##              119 
    ## [1] "mutation_location_x="
    ## [1] 119
    ## [1] "mutation_location_remove_dash="
    ## [1] 119
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CAT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 40
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CGT
    ## 1-letter AAString object
    ## seq: R
    ## [1] "MUTcodon and aa"
    ## $V174
    ## [1] "C"
    ## 
    ## $V175
    ## [1] "A"
    ## 
    ## $V176
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: H
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 22
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 46_T7 Biotec_CGN 
    ##              169 
    ## [1] "mutation_location_x="
    ## [1] 169
    ## [1] "mutation_location_remove_dash="
    ## [1] 169
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GAG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 57
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AAG
    ## 1-letter AAString object
    ## seq: K
    ## [1] "MUTcodon and aa"
    ## $V225
    ## [1] "G"
    ## 
    ## $V226
    ## [1] "A"
    ## 
    ## $V227
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: E
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 22
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 46_T7 Biotec_CGN 
    ##              384 
    ## [1] "mutation_location_x="
    ## [1] 384
    ## [1] "mutation_location_remove_dash="
    ## [1] 384
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: AGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 128
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AGT
    ## 1-letter AAString object
    ## seq: S
    ## [1] "MUTcodon and aa"
    ## $V438
    ## [1] "A"
    ## 
    ## $V439
    ## [1] "G"
    ## 
    ## $V440
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: S
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 23
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 51_T7 Biotec_CGN 
    ##              124 
    ## [1] "mutation_location_x="
    ## [1] 124
    ## [1] "mutation_location_remove_dash="
    ## [1] 124
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: AGT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 42
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TGT
    ## 1-letter AAString object
    ## seq: C
    ## [1] "MUTcodon and aa"
    ## $V180
    ## [1] "A"
    ## 
    ## $V181
    ## [1] "G"
    ## 
    ## $V182
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: S
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 23
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 51_T7 Biotec_CGN 
    ##              184 
    ## [1] "mutation_location_x="
    ## [1] 184
    ## [1] "mutation_location_remove_dash="
    ## [1] 184
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: ACT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 62
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GCT
    ## 1-letter AAString object
    ## seq: A
    ## [1] "MUTcodon and aa"
    ## $V240
    ## [1] "A"
    ## 
    ## $V241
    ## [1] "C"
    ## 
    ## $V242
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: T
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 23
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 51_T7 Biotec_CGN 
    ##              241 
    ## [1] "mutation_location_x="
    ## [1] 241
    ## [1] "mutation_location_remove_dash="
    ## [1] 241
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 81
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AGC
    ## 1-letter AAString object
    ## seq: S
    ## [1] "MUTcodon and aa"
    ## $V297
    ## [1] "G"
    ## 
    ## $V298
    ## [1] "G"
    ## 
    ## $V299
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: G
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 23
    ## [1] "j="
    ## [1] 4
    ## [1] "mutation_locations[i,j]="
    ## 51_T7 Biotec_CGN 
    ##              322 
    ## [1] "mutation_location_x="
    ## [1] 322
    ## [1] "mutation_location_remove_dash="
    ## [1] 322
    ## [1] "frame_shift_del"
    ## [1] "aaPosition"
    ## [1] 108
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: ATC
    ## 1-letter AAString object
    ## seq: I
    ## [1] "MUTcodon and aa"
    ## $V378
    ## [1] "-"
    ## 
    ## $V379
    ## [1] "T"
    ## 
    ## $V380
    ## [1] "C"
    ## 
    ## [1] "frame_shift_del"
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 23
    ## [1] "j="
    ## [1] 5
    ## [1] "mutation_locations[i,j]="
    ## 51_T7 Biotec_CGN 
    ##              594 
    ## [1] "mutation_location_x="
    ## [1] 594
    ## [1] "mutation_location_remove_dash="
    ## [1] 594
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: TAC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 198
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TAT
    ## 1-letter AAString object
    ## seq: Y
    ## [1] "MUTcodon and aa"
    ## $V648
    ## [1] "T"
    ## 
    ## $V649
    ## [1] "A"
    ## 
    ## $V650
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: Y
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 23
    ## [1] "j="
    ## [1] 6
    ## [1] "mutation_locations[i,j]="
    ## 51_T7 Biotec_CGN 
    ##              679 
    ## [1] "mutation_location_x="
    ## [1] 679
    ## [1] "mutation_location_remove_dash="
    ## [1] 679
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GAG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 227
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: AAG
    ## 1-letter AAString object
    ## seq: K
    ## [1] "MUTcodon and aa"
    ## $V735
    ## [1] "G"
    ## 
    ## $V736
    ## [1] "A"
    ## 
    ## $V737
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: E
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 25
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 53_T7 Biotec_CGN 
    ##               29 
    ## [1] "mutation_location_x="
    ## [1] 29
    ## [1] "mutation_location_remove_dash="
    ## [1] 29
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CTG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 10
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CCG
    ## 1-letter AAString object
    ## seq: P
    ## [1] "MUTcodon and aa"
    ## $V84
    ## [1] "C"
    ## 
    ## $V85
    ## [1] "T"
    ## 
    ## $V86
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: M
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 25
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 53_T7 Biotec_CGN 
    ##              456 
    ## [1] "mutation_location_x="
    ## [1] 456
    ## [1] "mutation_location_remove_dash="
    ## [1] 456
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CCT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 152
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CCC
    ## 1-letter AAString object
    ## seq: P
    ## [1] "MUTcodon and aa"
    ## $V510
    ## [1] "C"
    ## 
    ## $V511
    ## [1] "C"
    ## 
    ## $V512
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: P
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 25
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 53_T7 Biotec_CGN 
    ##              460 
    ## [1] "mutation_location_x="
    ## [1] 460
    ## [1] "mutation_location_remove_dash="
    ## [1] 460
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CAC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 154
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TAC
    ## 1-letter AAString object
    ## seq: Y
    ## [1] "MUTcodon and aa"
    ## $V516
    ## [1] "C"
    ## 
    ## $V517
    ## [1] "A"
    ## 
    ## $V518
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: H
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 25
    ## [1] "j="
    ## [1] 4
    ## [1] "mutation_locations[i,j]="
    ## 53_T7 Biotec_CGN 
    ##              579 
    ## [1] "mutation_location_x="
    ## [1] 579
    ## [1] "mutation_location_remove_dash="
    ## [1] 579
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GAG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 193
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GAA
    ## 1-letter AAString object
    ## seq: E
    ## [1] "MUTcodon and aa"
    ## $V633
    ## [1] "G"
    ## 
    ## $V634
    ## [1] "A"
    ## 
    ## $V635
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: E
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 26
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 54_T7 Biotec_CGN 
    ##              177 
    ## [1] "mutation_location_x="
    ## [1] 177
    ## [1] "mutation_location_remove_dash="
    ## [1] 177
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CCG
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 59
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: CCA
    ## 1-letter AAString object
    ## seq: P
    ## [1] "MUTcodon and aa"
    ## $V231
    ## [1] "C"
    ## 
    ## $V232
    ## [1] "C"
    ## 
    ## $V233
    ## [1] "G"
    ## 
    ## 1-letter AAString object
    ## seq: P
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 26
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 54_T7 Biotec_CGN 
    ##              326 
    ## [1] "mutation_location_x="
    ## [1] 326
    ## [1] "mutation_location_remove_dash="
    ## [1] 326
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GCT
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 109
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GTT
    ## 1-letter AAString object
    ## seq: V
    ## [1] "MUTcodon and aa"
    ## $V381
    ## [1] "G"
    ## 
    ## $V382
    ## [1] "C"
    ## 
    ## $V383
    ## [1] "T"
    ## 
    ## 1-letter AAString object
    ## seq: A
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 26
    ## [1] "j="
    ## [1] 3
    ## [1] "mutation_locations[i,j]="
    ## 54_T7 Biotec_CGN 
    ##              603 
    ## [1] "mutation_location_x="
    ## [1] 603
    ## [1] "mutation_location_remove_dash="
    ## [1] 603
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GGC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 201
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GGT
    ## 1-letter AAString object
    ## seq: G
    ## [1] "MUTcodon and aa"
    ## $V657
    ## [1] "G"
    ## 
    ## $V658
    ## [1] "G"
    ## 
    ## $V659
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: G
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 27
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 55_T7 Biotec_CGN 
    ##              229 
    ## [1] "mutation_location_x="
    ## [1] 229
    ## [1] "mutation_location_remove_dash="
    ## [1] 229
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CTC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 77
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TTC
    ## 1-letter AAString object
    ## seq: F
    ## [1] "MUTcodon and aa"
    ## $V285
    ## [1] "C"
    ## 
    ## $V286
    ## [1] "T"
    ## 
    ## $V287
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: L
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 27
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 55_T7 Biotec_CGN 
    ##              327 
    ## [1] "mutation_location_x="
    ## [1] 327
    ## [1] "mutation_location_remove_dash="
    ## [1] 327
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GTC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 109
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GTT
    ## 1-letter AAString object
    ## seq: V
    ## [1] "MUTcodon and aa"
    ## $V381
    ## [1] "G"
    ## 
    ## $V382
    ## [1] "T"
    ## 
    ## $V383
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: V
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 28
    ## [1] "j="
    ## [1] 1
    ## [1] "mutation_locations[i,j]="
    ## 56_T7 Biotec_CGN 
    ##              229 
    ## [1] "mutation_location_x="
    ## [1] 229
    ## [1] "mutation_location_remove_dash="
    ## [1] 229
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: CTC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 77
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: TTC
    ## 1-letter AAString object
    ## seq: F
    ## [1] "MUTcodon and aa"
    ## $V285
    ## [1] "C"
    ## 
    ## $V286
    ## [1] "T"
    ## 
    ## $V287
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: L
    ## [1] "-----END-----"
    ## [1] "-----start------"
    ## [1] "i="
    ## [1] 28
    ## [1] "j="
    ## [1] 2
    ## [1] "mutation_locations[i,j]="
    ## 56_T7 Biotec_CGN 
    ##              327 
    ## [1] "mutation_location_x="
    ## [1] 327
    ## [1] "mutation_location_remove_dash="
    ## [1] 327
    ## [1] "MUTcodon_DNAstr"
    ## 3-letter DNAString object
    ## seq: GTC
    ## [1] "translation!MUTaa"
    ## [1] "aaPosition"
    ## [1] 109
    ## [1] "WTcodon and aa"
    ## 3-letter DNAString object
    ## seq: GTT
    ## 1-letter AAString object
    ## seq: V
    ## [1] "MUTcodon and aa"
    ## $V381
    ## [1] "G"
    ## 
    ## $V382
    ## [1] "T"
    ## 
    ## $V383
    ## [1] "C"
    ## 
    ## 1-letter AAString object
    ## seq: V
    ## [1] "-----END-----"

show results: AAmutations\_df

    AAmutations_df

    ##                  [,1]                  [,2]    [,3]    [,4]                 
    ## ClyF             NA                    NA      NA      NA                   
    ## 11_T7 Biotec_CGN "N61D"                NA      NA      NA                   
    ## 12_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 13_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 14_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 15_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 21_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 22_T7 Biotec_CGN "G247frame_shift_ins" NA      NA      NA                   
    ## 23_T7 Biotec_CGN "A91A"                NA      NA      NA                   
    ## 24_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 25_T7 Biotec_CGN "Y198F"               NA      NA      NA                   
    ## 31_T7 Biotec_CGN "D35G"                "I210V" NA      NA                   
    ## 32_T7 Biotec_CGN "K57E"                "A62A"  "P85S"  "D122G"              
    ## 33_T7 Biotec_CGN "H5H"                 NA      NA      NA                   
    ## 34_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 35_T7 Biotec_CGN "F50S"                NA      NA      NA                   
    ## 41_T7 Biotec_CGN "P10M"                NA      NA      NA                   
    ## 42_T7 Biotec_CGN "S30S"                "R167R" "A181A" NA                   
    ## 43_T7 Biotec_CGN "G36G"                NA      NA      NA                   
    ## 44_T7 Biotec_CGN "G39G"                "C42C"  "V84E"  NA                   
    ## 45_T7 Biotec_CGN "K18R"                "G104R" NA      NA                   
    ## 46_T7 Biotec_CGN "R40H"                "K57E"  "S128S" NA                   
    ## 51_T7 Biotec_CGN "C42S"                "A62T"  "S81G"  "I108frame_shift_del"
    ## 52_T7 Biotec_CGN NA                    NA      NA      NA                   
    ## 53_T7 Biotec_CGN "P10M"                "P152P" "Y154H" "E193E"              
    ## 54_T7 Biotec_CGN "P59P"                "V109A" "G201G" NA                   
    ## 55_T7 Biotec_CGN "F77L"                "V109V" NA      NA                   
    ## 56_T7 Biotec_CGN "F77L"                "V109V" NA      NA                   
    ##                  [,5]    [,6]   
    ## ClyF             NA      NA     
    ## 11_T7 Biotec_CGN NA      NA     
    ## 12_T7 Biotec_CGN NA      NA     
    ## 13_T7 Biotec_CGN NA      NA     
    ## 14_T7 Biotec_CGN NA      NA     
    ## 15_T7 Biotec_CGN NA      NA     
    ## 21_T7 Biotec_CGN NA      NA     
    ## 22_T7 Biotec_CGN NA      NA     
    ## 23_T7 Biotec_CGN NA      NA     
    ## 24_T7 Biotec_CGN NA      NA     
    ## 25_T7 Biotec_CGN NA      NA     
    ## 31_T7 Biotec_CGN NA      NA     
    ## 32_T7 Biotec_CGN NA      NA     
    ## 33_T7 Biotec_CGN NA      NA     
    ## 34_T7 Biotec_CGN NA      NA     
    ## 35_T7 Biotec_CGN NA      NA     
    ## 41_T7 Biotec_CGN NA      NA     
    ## 42_T7 Biotec_CGN NA      NA     
    ## 43_T7 Biotec_CGN NA      NA     
    ## 44_T7 Biotec_CGN NA      NA     
    ## 45_T7 Biotec_CGN NA      NA     
    ## 46_T7 Biotec_CGN NA      NA     
    ## 51_T7 Biotec_CGN "Y198Y" "K227E"
    ## 52_T7 Biotec_CGN NA      NA     
    ## 53_T7 Biotec_CGN NA      NA     
    ## 54_T7 Biotec_CGN NA      NA     
    ## 55_T7 Biotec_CGN NA      NA     
    ## 56_T7 Biotec_CGN NA      NA

Show results: DNA mutation location (alignment location)

    mutation_locations

    ##                  [,1] [,2] [,3] [,4] [,5] [,6]
    ## ClyF               NA   NA   NA   NA   NA   NA
    ## 11_T7 Biotec_CGN  181   NA   NA   NA   NA   NA
    ## 12_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 13_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 14_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 15_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 21_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 22_T7 Biotec_CGN  741   NA   NA   NA   NA   NA
    ## 23_T7 Biotec_CGN  273   NA   NA   NA   NA   NA
    ## 24_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 25_T7 Biotec_CGN  593   NA   NA   NA   NA   NA
    ## 31_T7 Biotec_CGN  104  628   NA   NA   NA   NA
    ## 32_T7 Biotec_CGN  169  186  253  365   NA   NA
    ## 33_T7 Biotec_CGN   15   NA   NA   NA   NA   NA
    ## 34_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 35_T7 Biotec_CGN  149   NA   NA   NA   NA   NA
    ## 41_T7 Biotec_CGN   29   NA   NA   NA   NA   NA
    ## 42_T7 Biotec_CGN   90  501  543   NA   NA   NA
    ## 43_T7 Biotec_CGN  108   NA   NA   NA   NA   NA
    ## 44_T7 Biotec_CGN  117  126  251   NA   NA   NA
    ## 45_T7 Biotec_CGN   53  310   NA   NA   NA   NA
    ## 46_T7 Biotec_CGN  119  169  384   NA   NA   NA
    ## 51_T7 Biotec_CGN  124  184  241  322  594  679
    ## 52_T7 Biotec_CGN   NA   NA   NA   NA   NA   NA
    ## 53_T7 Biotec_CGN   29  456  460  579   NA   NA
    ## 54_T7 Biotec_CGN  177  326  603   NA   NA   NA
    ## 55_T7 Biotec_CGN  229  327   NA   NA   NA   NA
    ## 56_T7 Biotec_CGN  229  327   NA   NA   NA   NA

Show results: DNA mutation counts. -: no mutation AG: A to G GA: G to A
Del: Deletion Ins: Insertion ?: Not recognized nucleotide

    mutation_count_row

    ##                    - AG GA AT TA AC CA GC CG GT TG CT TC Del Ins ?
    ## ClyF             754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 11_T7 Biotec_CGN 753  1  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 12_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 13_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 14_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 15_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 21_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 22_T7 Biotec_CGN 753  0  0  0  0  0  0  0  0  0  0  0  0   0   1 0
    ## 23_T7 Biotec_CGN 753  1  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 24_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 25_T7 Biotec_CGN 753  0  0  1  0  0  0  0  0  0  0  0  0   0   0 0
    ## 31_T7 Biotec_CGN 752  2  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 32_T7 Biotec_CGN 750  2  0  0  0  0  0  0  0  0  0  1  1   0   0 0
    ## 33_T7 Biotec_CGN 753  0  0  0  0  0  0  0  0  0  0  0  1   0   0 0
    ## 34_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 35_T7 Biotec_CGN 753  0  0  0  0  0  0  0  0  0  0  0  1   0   0 0
    ## 41_T7 Biotec_CGN 753  0  0  0  0  0  0  0  0  0  0  1  0   0   0 0
    ## 42_T7 Biotec_CGN 751  0  1  0  0  0  0  0  0  0  0  1  1   0   0 0
    ## 43_T7 Biotec_CGN 753  1  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 44_T7 Biotec_CGN 751  0  0  0  1  0  0  0  0  0  0  0  2   0   0 0
    ## 45_T7 Biotec_CGN 752  1  1  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 46_T7 Biotec_CGN 751  1  1  0  0  0  0  0  0  0  0  0  1   0   0 0
    ## 51_T7 Biotec_CGN 748  2  1  0  1  0  0  0  0  0  0  0  1   1   0 0
    ## 52_T7 Biotec_CGN 754  0  0  0  0  0  0  0  0  0  0  0  0   0   0 0
    ## 53_T7 Biotec_CGN 750  1  0  0  0  0  0  0  0  0  0  2  1   0   0 0
    ## 54_T7 Biotec_CGN 751  1  0  0  0  0  0  0  0  0  0  0  2   0   0 0
    ## 55_T7 Biotec_CGN 752  0  0  0  0  0  0  0  0  0  0  0  2   0   0 0
    ## 56_T7 Biotec_CGN 752  0  0  0  0  0  0  0  0  0  0  0  2   0   0 0

export the result tables as csv

    ##turn "NA" to nothing for output csv
    AAmutations_removeNA <- AAmutations_df
    AAmutations_removeNA[is.na(AAmutations_removeNA)] <- ""
    mutation_locations_removeNA <- mutation_locations
    mutation_locations_removeNA[is.na(mutation_locations_removeNA)] <- ""
    ##create a new folder in the current working directory 
    dir.create("mutation_analysis_results", recursive = T)

    ## Warning in dir.create("mutation_analysis_results", recursive = T):
    ## 'mutation_analysis_results' already exists

    ##Save the dataframes 
    write.csv(mutation_count_row,"mutation_analysis_results/DNAmutation_counts.csv", row.names = TRUE)
    write.csv(mutation_locations_removeNA,"mutation_analysis_results/DNAmutation_locations.csv", row.names = TRUE)
    write.csv(AAmutations_removeNA,"mutation_analysis_results/AAmutations.csv", row.names = TRUE)

    list.files('mutation_analysis_results',full.names = T)

    ## [1] "mutation_analysis_results/AAmutations.csv"          
    ## [2] "mutation_analysis_results/DNAmutation_counts.csv"   
    ## [3] "mutation_analysis_results/DNAmutation_locations.csv"
